### PR TITLE
[Update] Add function for update

### PR DESCRIPTION
**Pull request by geshtng**
Create new function `Update` to update an article

### CODE REVIEW

File: handler/rest.go [hunk 1]
```diff
@@ -132,4 +132,37 @@ func (h ArticleHandler) Delete(c echo.Context) (err error) {
...
+	
+	dbRes, err := h.DB.Exec(query, item.Title, item.Body, articleID)
+	if err != nil {
+		resp := ErroResponse{
+			Message: err.Error(),
+		}
+		return c.JSON(http.StatusInternalServerError, resp)
+	}
+
+	updatedID, err := dbRes.RowsAffected()
```
Comment (bxcodec) [inline handler/rest.go]: This function won't return `updatedID`, but the total of items that updated.
So for update case, you only check for total updated items. 

So in line 166 you maybe you should change into like this.

```go
if totalUpdated <=0 {
  resp := ErroResponse{
	      Message: "Nothing is changed",
	    }
  return c.JSON(http.StatusInternalServerError, resp)
}

```

Comment (geshtng) [reply handler/rest.go]: Yeah I was confused about the output in postman before I update my code. The output:
```json
{
    "id": "1",
    "title": "UpdateKuy",
    "body": "Kuylah Update"
}
```
The action is `PUT: localhost:9090/articles/5` -> `ID` is `5`

After I update my code with above, the output show this:
```json
{
    "id": "",
    "title": "Kuy",
    "body": "Kuylah"
}
```
Not showing the updated article's `ID`

File: handler/rest.go [hunk 2]
```diff
@@ -132,4 +132,43 @@ func (h ArticleHandler) Delete(c echo.Context) (err error) {
...
+	articleID := c.Param("id")
+	var item models.Article
+	err = c.Bind(&item)
+	if err != nil {
+		resp := ErroResponse{
+			Message: err.Error(),
+		}
+		return c.JSON(http.StatusUnprocessableEntity, resp)
+	}
+
```
Comment (bxcodec) [inline handler/rest.go]: After binding the value, re-assign the ID.

`item.ID = articleID`

File: handler/rest.go (hunk 1 above)
Comment (bxcodec) [reply handler/rest.go]: Should be commented in the next review

File: handler/rest.go (hunk 2 above)
Comment (geshtng) [reply handler/rest.go]: Done

File: handler/rest.go [hunk 3]
```diff
@@ -132,4 +132,45 @@ func (h ArticleHandler) Delete(c echo.Context) (err error) {
...
+	if totalUpdated <= 0 {
+		resp := ErroResponse{
+			Message: "Nothing is changed",
+		}
+		return c.JSON(http.StatusInternalServerError, resp)
+	}
+	
+	item.ID = articleID
```
Comment (bxcodec) [inline handler/rest.go]: Saran sih, ID nya di assign di atas aja, di line 147. Alasannya, validasi semua value dulu baru lakukan bisnis rule, termasuk eksekusi di DB. 

Coba bayangin, semua fungsi yang berhubungan DB kau pisahkan ke package berbeda, fungsi berbeda. Untuk dia lebih clean dan gadak redundant, mending pengisian ID ada di atas. Minor error (human error, karena lupa assgin ID).

Comment (geshtng) [reply handler/rest.go]: Terima kasih sarannya bang :+1:

**Comment by bxcodec** [comment]
LGTM

**Comment by geshtng** [comment]
Ngga di-Accept dulu bang? 😂😂

**Pull request merged**